src/main/java/edu/cornell/kfs/fp/document/RecurringDisbursementVoucherDocument.java

```diff
@@ -0,0 +1,138 @@
+
+import java.sql.Date;
+import java.util.Calendar;
+import java.util.Iterator;
+import java.util.TreeMap;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.businessobject.AccountingLine;
+import org.kuali.kfs.sys.businessobject.GeneralLedgerPendingEntry;
+import org.kuali.kfs.sys.businessobject.GeneralLedgerPendingEntrySequenceHelper;
+import org.kuali.kfs.sys.businessobject.GeneralLedgerPendingEntrySourceDetail;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.kfs.sys.service.GeneralLedgerPendingEntryService;
+import org.kuali.kfs.sys.service.HomeOriginationService;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+import org.kuali.rice.coreservice.framework.parameter.ParameterConstants.COMPONENT;
+import org.kuali.rice.coreservice.framework.parameter.ParameterConstants.NAMESPACE;
+import org.kuali.rice.krad.exception.ValidationException;
+import org.kuali.rice.krad.util.ObjectUtils;
+
+import edu.cornell.kfs.fp.businessobject.ScheduledAccountingLine;
+import edu.cornell.kfs.fp.businessobject.ScheduledSourceAccountingLine;
+import edu.cornell.kfs.fp.service.impl.ScheduledAccountingLineServiceImpl;
+import edu.cornell.kfs.gl.service.ScheduledAccountingLineService;
+
+@NAMESPACE(namespace = KFSConstants.CoreModuleNamespaces.FINANCIAL)
+@COMPONENT(component = "RecurringDisbursementVoucher")
+public class RecurringDisbursementVoucherDocument extends CuDisbursementVoucherDocument {
+
+	private static final long serialVersionUID = 5578411133159987973L;
+	protected static Log LOG = LogFactory.getLog(RecurringDisbursementVoucherDocument.class);
+	private ScheduledAccountingLineService scheduledAccountingLineService;
+	
+	@Override
+    public void customizeExplicitGeneralLedgerPendingEntry(GeneralLedgerPendingEntrySourceDetail postable, GeneralLedgerPendingEntry explicitEntry) {
+		super.customizeExplicitGeneralLedgerPendingEntry(postable, explicitEntry);
+		ScheduledAccountingLine accountingLine = (ScheduledAccountingLine) postable;
+        explicitEntry.setTransactionEntryProcessedTs(null);
+        if (accountingLine.isSourceAccountingLine()) {
+            explicitEntry.setReferenceFinancialSystemOriginationCode(getHomeOriginationService().getHomeOrigination().getFinSystemHomeOriginationCode());
+            explicitEntry.setReferenceFinancialDocumentNumber(accountingLine.getReferenceNumber());
+            explicitEntry.setReferenceFinancialDocumentTypeCode(explicitEntry.getFinancialDocumentTypeCode());
+        } else if (accountingLine.isTargetAccountingLine()) {
+        	throw new IllegalArgumentException("we shouldn't have target lines on DVs");
+        }
+    }
+	
+	@Override
+    protected void processExplicitGeneralLedgerPendingEntry(GeneralLedgerPendingEntrySequenceHelper sequenceHelper,
+        GeneralLedgerPendingEntrySourceDetail glpeSourceDetail, GeneralLedgerPendingEntry explicitEntry) {
```

Comment:
I would still like to see this large method refactored into smaller methods, each doing one thing (as per Clean Code).